src/main/java/com/team701/buddymatcher/socket/ChatEventHandler.java

```diff
@@ -34,37 +48,115 @@ public ChatEventHandler(SocketIOServer server, MessageRepository messageReposito
         this.namespace.addConnectListener(onConnected());
         this.namespace.addDisconnectListener(onDisconnected());
         this.namespace.addEventListener(MESSAGE_EVENT_KEY, MessageObject.class, onMessageReceived());
+        this.namespace.addEventListener(READ_EVENT_KEY, ReadObject.class, onReadReceived());
+
+        this.messageRepository = messageRepository;
     }
 
+    /**
+     * onMessageReceived listener that sends message to the receiver when it receives a message event
+     * @return
+     */
     private DataListener<MessageObject> onMessageReceived() {
         return (client, data, ackSender) -> {
-            log.debug("Client[{}] - Received chat message '{}'", client.getSessionId().toString(), data);
+            log.debug("Client[{}] - Received message '{}'", client.getSessionId().toString(), data);
 
-            SocketIOClient receiver = namespace.getClient(userConnections.get(data.receiverId));
-            if (receiver != null) receiver.sendEvent(MESSAGE_EVENT_KEY, data);
+            SocketIOClient receiver = namespace.getClient(userIdConnections.get(data.getReceiverId()));
+            if (receiver != null) sendMessage(receiver, data);
             putMessageInHistory(data);
         };
     }
 
+    /**
+     * onReadReceived listener that sends read notice to the buddy when it receives a read event
+     * @return
+     */
+    private DataListener<ReadObject> onReadReceived() {
+        return (client, data, ackSender) -> {
```

Comment:
Why are you returning a lambda here (and in other places)? I think it could be a bit clearer by changing it to something like `onReadReceived(client, data, ackSender)` and then where it's used do `addEventListener(key, clazz, onMessageReceived);` (I might be forgetting the exact syntax for Java)

Reply:
straight up copied from https://github.com/jamesjieye/netty-socketio.spring/blob/master/src/main/java/com/jamesye/starter/realtimeserver/modules/chat/ChatModule.java#L29-L34

No idea if it actually works